### ISSUE TITLE
Error trying to run bundle lock --update undefined

### ISSUE THREAD

**Issue by mrsimo** [renovatebot/renovate#15949]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.78.1

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

* Renovate is trying to upgrade the `rails` gem from `6.0.5` to `6.1.6` using `rangeStrategy=update-lockfile`.
* We have another gem, `roadie-rails`, defined in the `Gemfile` with no version specified, but currently on [`2.1.1`](https://rubygems.org/gems/roadie-rails/versions/2.1.1) in the `Gemfile.lock`. This version defines `rails < 6.1`.
* The upgrade fails because of `roadie-rails`, as expected, but on the pull request it adds a comment with "Artifact update problem" that specifies:

File name: Gemfile.lock

```
Could not find gem 'undefined'.
```

I would expect `roadie-rails` to be mentioned somewhere, as seen when I try to run the same myself:

```
❯ bundler lock --update rails
Fetching gem metadata from https://rubygems.org/...........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails (= 6.1.6) was resolved to 6.1.6, which depends on
      railties (= 6.1.6)

    roadie-rails was resolved to 2.1.1, which depends on
      railties (>= 5.1, < 6.1)
```

The logs seem to show that renovate detects a problem updating rails, even finds that `roadie-rails` is the problem but then tries to do a `bundler lock --update rails undefined undefined`.

Here's the minimal reproduction: https://github.com/mrsimo/renovate-bug-undefined

To double check that it wasn't a problem with two private gems we have I removed them, but the problem still occurs in that repository as you might see [from the two PR's my test run created](https://github.com/mrsimo/renovate-bug-undefined/pulls).

### Relevant debug logs

I think this is the relevant bits. I redacted the `updatedDeps` from the `Found new resolve matches - reattempting recursively` because it didn't seem to have anything useful. That line does say `"resolveMatches":["rails","roadie-rails"]`.

<details><summary>Logs</summary>

```json
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","command":"bundler lock --update rails","msg":"Executing command","time":"2022-06-08T15:30:51.817Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","err":{"code":6,"killed":false,"signal":null,"cmd":"bundler lock --update rails","stdout":"Fetching https://**redacted**@github.com/harvesthq/harvest-revision.git\nFetching https://**redacted**@github.com/harvesthq/harvest-lb-check.git\nFetching gem metadata from https://rubygems.org/..........\nResolving dependencies....\n","stderr":"Bundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n","message":"Command failed: bundler lock --update rails\nBundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n","stack":"Error: Command failed: bundler lock --update rails\nBundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n\n    at ChildProcess.exithandler (node:child_process:398:12)\n    at ChildProcess.emit (node:events:527:28)\n    at ChildProcess.emit (node:domain:475:12)\n    at maybeClose (node:internal/child_process:1092:16)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)"},"msg":"rawExec err","time":"2022-06-08T15:30:56.204Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","err":{"code":6,"killed":false,"signal":null,"cmd":"bundler lock --update rails","stdout":"Fetching https://**redacted**@github.com/harvesthq/harvest-revision.git\nFetching https://**redacted**@github.com/harvesthq/harvest-lb-check.git\nFetching gem metadata from https://rubygems.org/..........\nResolving dependencies....\n","stderr":"Bundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n","message":"Command failed: bundler lock --update rails\nBundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n","stack":"Error: Command failed: bundler lock --update rails\nBundler could not find compatible versions for gem \"railties\":\n  In Gemfile:\n    rails (= 6.1.6) was resolved to 6.1.6, which depends on\n      railties (= 6.1.6)\n\n    roadie-rails was resolved to 2.1.1, which depends on\n      railties (>= 5.1, < 6.1)\n\n    at ChildProcess.exithandler (node:child_process:398:12)\n    at ChildProcess.emit (node:events:527:28)\n    at ChildProcess.emit (node:domain:475:12)\n    at maybeClose (node:internal/child_process:1092:16)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)"},"msg":"Bundler has a resolve error","time":"2022-06-08T15:30:56.205Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","resolveMatches":["rails","roadie-rails"],"updatedDeps":["**redacted**"],"msg":"Found new resolve matches - reattempting recursively","time":"2022-06-08T15:30:56.205Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","msg":"bundler.updateArtifacts(Gemfile)","time":"2022-06-08T15:30:56.209Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","msg":"Using bundler version specified in lockfile","time":"2022-06-08T15:30:56.210Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","msg":"Using ruby version specified in .ruby-version","time":"2022-06-08T15:30:56.211Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","command":"bundler lock --update rails undefined undefined","msg":"Executing command","time":"2022-06-08T15:30:56.212Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":20,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","err":{"code":7,"killed":false,"signal":null,"cmd":"bundler lock --update rails undefined undefined","stdout":"","stderr":"Could not find gem 'undefined'.\n","message":"Command failed: bundler lock --update rails undefined undefined\nCould not find gem 'undefined'.\n","stack":"Error: Command failed: bundler lock --update rails undefined undefined\nCould not find gem 'undefined'.\n\n    at ChildProcess.exithandler (node:child_process:398:12)\n    at ChildProcess.emit (node:events:527:28)\n    at ChildProcess.emit (node:domain:475:12)\n    at maybeClose (node:internal/child_process:1092:16)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)"},"msg":"rawExec err","time":"2022-06-08T15:30:56.549Z","v":0}
{"name":"renovate","hostname":"renovate-27578370-pc7dt","pid":8,"level":30,"logContext":"8Hq1HOVjZdpBW_sRpYtam","repository":"harvesthq/paperboy","branch":"renovate-rails-6.x","err":{"code":7,"killed":false,"signal":null,"cmd":"bundler lock --update rails undefined undefined","stdout":"","stderr":"Could not find gem 'undefined'.\n","message":"Command failed: bundler lock --update rails undefined undefined\nCould not find gem 'undefined'.\n","stack":"Error: Command failed: bundler lock --update rails undefined undefined\nCould not find gem 'undefined'.\n\n    at ChildProcess.exithandler (node:child_process:398:12)\n    at ChildProcess.emit (node:events:527:28)\n    at ChildProcess.emit (node:domain:475:12)\n    at maybeClose (node:internal/child_process:1092:16)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)"},"msg":"Gemfile.lock update failed due to an unknown reason","time":"2022-06-08T15:30:56.550Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by ajplarson** [comment]
+1 on 'Could not find gem undefined ' having similar error
`Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rack (~> 3.0)

    thin (~> 1.8) was resolved to 1.8.1, which depends on
      rack (< 3, >= 1)`

**Comment by viceice** [comment]
@ajplarson which version of renovate are you using?

**Comment by viceice** [comment]
- fixes in #16462

**Comment by viceice** [comment]
@ajplarson please open a new discussion.
